pkg/geo/geomfn/generate_points.go

```diff
@@ -0,0 +1,227 @@
+	area, err := Area(g)
+	if err != nil {
+		return nil, errors.Newf("could not calculate Polygon area: %v", err)
+	}
+	bbox := g.CartesianBoundingBox()
+	bboxWidth := bbox.HiX - bbox.LoX
+	bboxHeight := bbox.HiY - bbox.LoY
+	bboxArea := bboxHeight * bboxWidth
+	if area == 0.0 || bboxArea == 0.0 {
+		return nil, errors.New("zero area input Polygon")
+	}
+	// Gross up our test set a bit to increase odds of getting coverage in one pass.
+	sampleNpoints := float64(npoints) * bboxArea / area
+
+	sampleSqrt := math.Round(math.Sqrt(sampleNpoints))
+	if sampleSqrt == 0 {
+		sampleSqrt = 1
+	}
+	var sampleHeight, sampleWidth int
+	var sampleCellSize float64
+	// Calculate the grids we're going to randomize within.
+	if bboxWidth > bboxHeight {
+		sampleWidth = int(sampleSqrt)
+		sampleHeight = int(math.Ceil(sampleNpoints / float64(sampleWidth)))
+		sampleCellSize = bboxWidth / float64(sampleWidth)
+	} else {
+		sampleHeight = int(sampleSqrt)
+		sampleWidth = int(math.Ceil(sampleNpoints / float64(sampleHeight)))
+		sampleCellSize = bboxHeight / float64(sampleHeight)
+	}
+	// Prepare the polygon for fast true/false testing.
+	gprep, err := geos.PrepareGeometry(g.EWKB())
+	if err != nil {
+		return nil, errors.Newf("could not prepare Geometry: %v", err)
+	}
+	defer func() {
+		err = geos.PreparedGeomDestroy(gprep)
+	}()
+
+	// Initiate random number generator.
+	randomNumberGenerator := rand.New(rand.NewSource(seed))
+	// Generate a slice of points - for every cell on a grid store coordinates.
+	n := sampleHeight * sampleWidth
+	cells := make([]geom.Coord, n)
+	for i := 0; i < sampleWidth; i++ {
+		for j := 0; j < sampleHeight; j++ {
+			cells[i*int(sampleHeight)+j] = geom.Coord{float64(i), float64(j)}
+		}
+	}
+	// Shuffle the points. Without shuffling, the generated point will
+	// always be adjacent to the previous one (in terms of grid cells).
+	if n > 1 {
+		randomNumberGenerator.Shuffle(n, func(i int, j int) {
+			temp := cells[j]
+			cells[j] = cells[i]
+			cells[i] = temp
+		})
+	}
+	results := geom.NewMultiPoint(geom.XY)
+	// Generate points and test them.
+	npointsGenerated := 0
+	iterations := 0
+	for npointsGenerated < npoints {
+		iterations++
+		for _, cell := range cells {
+			y := bbox.LoY + cell.X()*sampleCellSize
+			x := bbox.LoX + cell.Y()*sampleCellSize
+			x += randomNumberGenerator.Float64() * sampleCellSize
+			y += randomNumberGenerator.Float64() * sampleCellSize
+			if x > bbox.HiX || y > bbox.HiY {
+				continue
+			}
+			gseq, err := geos.CoordSequenceCreate(1, 2)
+			if err != nil {
+				return nil, errors.Newf("could not create coord sequence: %v", err)
+			}
+			gseq, err = geos.CoordSequenceSetXY(gseq, 0, x, y)
+			if err != nil {
+				return nil, errors.Newf("could not set X and Y for coord sequence: %v", err)
+			}
+			defer func() {
+				// err = geos.CoordSequenceDestroy(gseq)
+			}()
+			gpt, err := geos.GeometryPointCreate(gseq)
```

Comment:
Does `gpt` need to be destroyed explicitly?

Reply:
I would think so.